app/controllers/admin/listing_shapes_controller.rb

```diff
@@ -344,14 +341,57 @@ def params_to_shape(params)
     def shape_to_locals(shape)
       shape = Shape.call(shape)
 
+      units = split_availability_and_pricing_units(shape)
+
       shape.merge(
-        predefined_units: expand_predefined_units(shape[:units]),
-        custom_units: encode_custom_units(shape[:units].select { |unit| unit[:type] == :custom })
+        availability_unit: units[:availability],
+        predefined_units: expand_predefined_units(units[:pricing]),
+        custom_units: encode_custom_units(units[:pricing].select { |unit| unit[:type] == :custom })
       )
     end
 
     # private
 
+    # Splits units to availability units and pricing units.
+    #
+    # In the backend we have only one concept, units. However, in the UI
+    # we are showing pricing unit checkboxes and availability unit radio
+    # buttons. This method maps backend units to the two unit formats
+    # we have in the UI
+    #
+    def split_availability_and_pricing_units(shape)
+      if shape[:availability] == :booking
+        {
+          pricing: [],
+          availability: shape[:units].first[:type]
+        }
+      else
+        {
+          pricing: shape[:units],
+          availability: nil
+        }
+      end
+    end
+
+    # Combines availability units and pricing units to just "units".
+    #
+    # In the backend we have only one concept, units. However, in the UI
+    # we are showing pricing unit checkboxes and availability unit radio
+    # buttons. This method maps the UI units to backend units.
+    #
+    def parse_units_from_params(form_params)
+      selected_predefined_units =
+        if form_params[:availability] == "booking"
```

Comment:
Ideally it should be in different order so that the "logic" can always consistently deal with symbols. Now it adds a bit of complexity that mapping to one way uses symbols and to other strings. But it's not a big deal.